src/remat.jl

```diff
@@ -386,13 +386,22 @@ end
 
 rmulΛ!(A::M, B::ReMat{T,1}) where{M<:AbstractMatrix{T}} where{T} = rmul!(A, first(B.λ))
 
-function rmulΛ!(A::M, B::ReMat{T,S}) where{M<:AbstractMatrix{T}} where{T,S}
+function rmulΛ!(A::M, B::ReMat{T,S}) where {M<:AbstractMatrix{T}} where {T,S}
     m, n = size(A)
     q, r = divrem(n, S)
-    iszero(r) || throw(DimensionMismatch("size(A, 2) = is not a multiple of block size"))
-    A3 = reshape(A, (m, S, q))
-    for k in 1:q
-        rmul!(view(A3, :, :, k), B.λ)
+    iszero(r) || throw(DimensionMismatch("size(A, 2) is not a multiple of block size"))
+    Bd = B.λ.data
+    @inbounds for k = 1:q
```

Comment:
I like the comment below on which multiplication these loops are replacing. Can you do the same here?